src/utils/mutations.ts

```diff
@@ -277,6 +286,83 @@ export async function updatePlayer(
 	})
 }
 
+// Mutation to update Forum Post
+export async function updateForumPost(
+	updatedPost: {
+		title: string
+		content: string
+		postType: number
+	},
+	classroomID: string,
+	postID: string,
+) {
+	const postRef = doc(db, `classrooms/${classroomID}/forumPosts/${postID}`)
+	await updateDoc(postRef, {
+		title: updatedPost.title,
+		content: updatedPost.content,
+		postType: updatedPost.postType,
+	})
+}
+
+export async function updateForumPostLikes(
+	classroomID: string,
+	postID: string,
+	likerID: string,
+	add: boolean,
+) {
+	const postRef = doc(db, `classrooms/${classroomID}/forumPosts/${postID}`)
+	if (add) {
+		await updateDoc(postRef, {
+			likes: increment(1),
+			likers: arrayUnion(likerID),
+		})
+	} else {
+		await updateDoc(postRef, {
+			likes: increment(-1),
+			likers: arrayRemove(likerID),
+		})
+	}
+}
+
+export async function updateForumCommentLikes(
+	classroomID: string,
+	postID: string,
+	commentID: string,
+	likerID: string,
+	add: boolean,
+) {
+	const postRef = doc(db, `classrooms/${classroomID}/forumPosts/${postID}/comments/${commentID}`)
+	if (add) {
+		await updateDoc(postRef, {
+			likes: increment(1),
```

Comment:
I know this isn't reflected in our DB diagram yet but maybe it would be better to only store the list of likers and just calculate the likes based on that client-side?